compose.py

```diff
@@ -150,41 +151,79 @@ def create_argmap(images):
     # Extract information from upstream build and create map other functions can
     # use
     upstreamDockerImage = images["full"]
+    minDockerImage = images["min"]
+    enable_gpu = FLAGS.enable_gpu
+    # Docker inspect enviroment variables
+    baseRunArgs = ['docker', 'inspect', '-f']
+    import re  # parse all PATH enviroment variables
 
-    # first pull docker image
+    # first pull docker images
     log("pulling container:{}".format(upstreamDockerImage))
     p = subprocess.run(['docker', 'pull', upstreamDockerImage])
     fail_if(
         p.returncode != 0,
         'docker pull container {} failed, {}'.format(upstreamDockerImage,
                                                      p.stderr))
-
-    baseRunArgs = ['docker', 'inspect', '-f']
-    p_version = subprocess.run(baseRunArgs + [
+    if enable_gpu:
+        pm = subprocess.run(['docker', 'pull', minDockerImage])
+        fail_if(
+            pm.returncode != 0, 'docker pull container {} failed, {}'.format(
+                minDockerImage, pm.stderr))
+        pm_path = subprocess.run(baseRunArgs + [
+            '{{range $index, $value := .Config.Env}}{{$value}} {{end}}',
+            minDockerImage
+        ],
+                                 capture_output=True,
+                                 text=True)
+        fail_if(
+            pm_path.returncode != 0,
+            'docker inspect to find triton enviroment variables for min container failed, {}'
+            .format(pm_path.stderr))
+        # min container needs to be GPU enabled if the build is GPU build
+        vars = pm_path.stdout
+        e = re.search("CUDA_VERSION", vars)
+        gpu_enabled = False if e == None else True
+        fail_if(
+            not gpu_enabled,
+            '\'enable-gpu\' flag added but min container provided does not have CUDA installed'
```

Comment:
Instead of added say specified.

Reply:
Please address changes here and in other places

Reply:
were you looking at a old version? I _think_ I addressed all your comments

Reply:
Yes, I was. Looks like you address all the changes